packages/playwright-core/src/server/frames.ts

```diff
@@ -471,6 +471,7 @@ export class Frame extends SdkObject {
     this._stopNetworkIdleTimer();
     if (this._inflightRequests.size === 0)
       this._startNetworkIdleTimer();
+    this._onLifecycleEvent('commit');
```

Comment:
Should we also do this once in the constructor?

Reply:
It already works on initial url. Added a test.

Reply:
As discussed offline fire 'commit' event in the constructor just in case.